persistent-sqlite/Database/Persist/Sqlite.hs

```diff
@@ -561,30 +567,38 @@ getCopyTable allDefs getter def = do
                 return $ name : names
             Just y -> error $ "Invalid result from PRAGMA table_info: " ++ show y
     table = getEntityDBName def
-    tableTmp = EntityNameDB $ unEntityNameDB table <> "_backup"
+    schema = getEntitySchema def
+    tableIdentifier = entityIdentifier def
+    -- Temporary tables cannot have qualified names, so we prepend
+    -- the name with the intended schema to provide namespacing.
+    tableTmp =
+        EntityNameDB $ T.intercalate "_" $ catMaybes [unSchemaNameDB <$> schema, Just $ unEntityNameDB table, Just "backup"]
```

Comment:
Is the philosophy here that Persistent will provide ersatz schema support by using a name prefix here, even though sqlite itself doesn't allow schemas for temp tables?

I can imagine that's maybe defensible, but ideally Persistent would just reflect what the underlying implementation supports, even in partial cases like this (i.e. you need to know as an implementor that sqlite won't support this, so if you try adding it in Persistent, it will throw an error or something).

I don't know Persistent's general philosophy though, so I might be off base there.

Also, I don't really know how you get Persistent to interface with a temporary table anyway, and even then this feels like a corner case...